src/Components/NewsletterUpdate/NewsletterUpdateContainer.js

```diff
@@ -0,0 +1,201 @@
+import React, { createContext, useEffect } from "react";
+import useReducerWithSideEffects, {
+  UpdateWithSideEffect,
+  Update
+} from "use-reducer-with-side-effects";
+import {
+  SHOW_ALERT,
+  LOADING,
+  HANDLE_SUBMIT,
+  SET_SELECT,
+  GET_NEWSLETTERS_FETCH,
+  GET_NEWSLETTERS_SUCCESS,
+  SWITCH_TO_UPDATE
+} from "../../utils/action-types";
+import { getErrorMessages } from "../common/Helpers";
+
+const initialState = {
+  didSubToNewslettersBefore: false,
+  newsletters: [],
+  isListLoading: true,
+  isSubmitting: false,
+  alert: {
+    type: "error",
+    content: ""
+  }
+};
+const store = createContext(initialState);
+const { Provider } = store;
+
+const NewsletterUpdateContainer = ({
+  style,
+  className = "",
+  onSuccess = () => {},
+  onFailure = () => {},
+  children
+}) => {
+  const handleSubmit = (
+    { newsletters, didSubToNewslettersBefore },
+    dispatch
+  ) => {
+    const callback = (err, res) => {
+      dispatch({ type: LOADING, payload: false });
+
+      if (err) {
+        dispatch({
+          type: SHOW_ALERT,
+          payload: {
+            type: "error",
+            content: getErrorMessages(err)
+          }
+        });
+        onFailure(err);
+      } else {
+        if (!didSubToNewslettersBefore) {
+          dispatch({ type: SWITCH_TO_UPDATE });
+        }
+        onSuccess(res);
+      }
+    };
+
+    const requestData = {
+      email: window.Pelcro.user.read()?.email,
+      source: "web",
+      lists: newsletters
+        .filter((newsletter) => newsletter.selected)
+        .map((newsletter) => newsletter.id)
+        .join(",")
+    };
+
+    if (didSubToNewslettersBefore) {
+      window.Pelcro.newsletter.update(requestData, callback);
```

Comment:
So update doesn't work if it's the first newsletter?

Reply:
update would fail if this is the first newsletter you're subbing to